lib/features/public_asset/domain/usecase/remove_identity_from_public_assets_interactor.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:jmap_dart_client/jmap/mail/extensions/public_asset/public_asset.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/repository/public_asset_repository.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/remove_identity_from_public_assets_state.dart';
+import 'package:tmail_ui_user/features/public_asset/presentation/public_asset_controller.dart';
+
+class RemoveIdentityFromPublicAssetsInteractor {
+  RemoveIdentityFromPublicAssetsInteractor(this._publicAssetRepository);
+
+  final PublicAssetRepository _publicAssetRepository;
+
+  Stream<Either<Failure, Success>> execute(
+    Session session,
+    AccountId accountId,
+    {
+      required IdentityId identityId,
+      required List<PublicAssetId> publicAssetIds
+    }
+  ) async* {
+    try {
+      yield Right(RemovingIdentityFromPublicAssetsState());
+      final publicAssets = await _publicAssetRepository.getPublicAssetsFromIds(
+        session,
+        accountId,
+        publicAssetIds: publicAssetIds
+      );
+      if (publicAssets.isEmpty) {
+        yield Left(RemoveIdentityFromPublicAssetsFailureState());
```

Comment:
IMO, should have a dedicated error state for this: `NotFoundAnyPublicAssets`

Reply:
Done